PyriteEditor/src/editor/bridges/Lights/pf_Light.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include "world/Lights/Light.h"
+
+namespace pye
+{
+	// pyr format
+
+	struct pf_Light
+	{
+		
+		// -- // 
+		pyr::BaseLight* sourceLight;
+
+	public:
+		enum Type { POINT, DIR, SPOT, NONE };
+		// -- //
+		std::string name;
+		Type type;
+	};
+
+	struct pf_LightsCollection
+	{
+		pf_LightsCollection() {}
+		pf_LightsCollection(pyr::LightsCollections* source)
+		{
+			assert(source);
+			sourceCollection = source;
+			UpdateEditorFormat();
+		}
+		// -- // 
+		pyr::LightsCollections* sourceCollection;
+
+
+		// -- // 
+		std::vector<pf_Light> EditorFormat;
+		int selectedId = 0;
+		pf_Light* selectedLight = nullptr;
+
+	public:
+
+		void UpdateEditorFormat()
```

Comment:
Not a fan of having this here, this class is half-data and half-logic

Reply:
Not entirely sure i agree on this one. The goal is to have this half data / half logic, which is basically a model-view-view-model design